graphql/resolvers/repo.go

```diff
@@ -149,13 +150,12 @@ func (repoResolver) Identity(_ context.Context, obj *models.Repository, prefix s
 	return models.NewLazyIdentity(obj.Repo, excerpt), nil
 }
 
-func (repoResolver) UserIdentity(_ context.Context, obj *models.Repository) (models.IdentityWrapper, error) {
-	excerpt, err := obj.Repo.GetUserIdentityExcerpt()
-	if err != nil {
-		return nil, err
+func (repoResolver) UserIdentity(ctx context.Context, obj *models.Repository) (models.IdentityWrapper, error) {
+	id := identity.ForContext(ctx, obj.Repo)
+	if id == nil {
+		return nil, nil
```

Comment:
Shouldn't that be `ErrNotAuthenticated` as well ?

Reply:
IMO no: this isn't an error case - we're explicitly telling you that you don't have an identity at the moment, rather than informing you that you've tried to access an endpoint for which authentication is required.

Reply:
I've changed my mind; it makes most of the call-sites more convenient.